docs/caching_and_memoization.rst

```diff
@@ -238,17 +238,14 @@ memoize rather than quickcache. You could split it apart like this:
 .. code-block:: python
 
     @memoized
-    def _count_users_forms_by_device(self, domain, device_ids):
-        return {
-            FormAccessors(domain.name).count_forms_by_device(device_id)
-            for device_id in device_ids
-        }
+    def _count_users_forms_by_device(self, domain, device_id):
+        return FormAccessors(domain).count_forms_by_device(device_id)
 
     def count_users_forms_by_device(self, domain_obj, user):
-        return self._count_users_forms_by_device(
-            domain_obj.name,
-            [device.device_id for device in user.devices]
-        )
+        return {
+            self._count_users_forms_by_device(domain_obj.name, device.device_id)
+            for device in user.devices
+        }
```

Comment:
I'm still not thrilled with this example - it's not clear that `_count_users_forms_by_device` would be called multiple times.  Let me know if you've got any ideas for a better example of a function that:
* Should be cached or memoized
* Accepts objects as args
* Maybe one of those objects could be simplified (`domain_obj` vs `domain`)
* Another object _is_ needed (for properties or methods on it), but you could vary the cache on a simple property

Reply:
I guess the [`frozenset` suggestion](https://github.com/dimagi/commcare-hq/pull/20188#discussion_r180907313) doesn't meet that criteria? Or maybe it's too esoteric because `quickcache` doesn't support `frozenset`?

Reply:
It could also pass in a tuple, but I thought this was a better way of structuring it anyways.